consensus/ibft/snapshot.go

```diff
@@ -531,14 +531,39 @@ func readDataStore(path string, obj interface{}) error {
 }
 
 // writeDataStore attempts to write the specific file to file storage
-func writeDataStore(path string, obj interface{}) error {
+func writeDataStore(dir, filename string, obj interface{}) error {
 	data, err := json.Marshal(obj)
 	if err != nil {
 		return err
 	}
 
-	//nolint:gosec
-	if err := ioutil.WriteFile(path, data, 0755); err != nil {
+	// atomically overwrite file
+	// https://stackoverflow.com/questions/30385225/is-there-an-os-independent-way-to-atomically-overwrite-a-file
+	// create a temporary file
+	tmpFile, err := os.CreateTemp(dir, filename+"*.tmp")
```

Comment:
What should we do if we create `tmp` file failed?
e.g. `file already exists?`

Reply:
failure condition:
  1. disk full

`os.CreateTemp` never create the same filename